pkg/kubelet/lifecycle/handlers.go

```diff
@@ -149,6 +150,11 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 
 			// clear err since the fallback succeeded
 			if httpErr == nil {
+				// TODO: increment a metric about the fallback
+				if eventRecorder != nil {
+					// report the fallback with an event
+					eventRecorder.Event(pod, v1.EventTypeWarning, "LifecycleHTTPFallback", fmt.Sprintf("request to HTTPS lifecycle hook %s got HTTP response, retry with HTTP succeeded", req.URL.Host))
```

Comment:
Should the log at https://github.com/kubernetes/kubernetes/pull/113175/commits/122b43037eccc0c17ecd912e22d8e9f92a091040#diff-66f9167fad0ae80b15631b755bc6f6b3ce9324eec823ec01dd47d257707e9870R144 be changed to WARN level to maintain consistency with the event type?

Reply:
there's no WarnS, only InfoS and ErrorS